syntax = "proto3";

option go_package = "grpc.test/src/proto";

package grpc_demo;

service Test {
    rpc SendRequest (TestRequest) returns (TestReply) {}
    // rpc ServerStream (TestRequest) returns (stream TestReply) {}
    rpc ClientStream (stream NumberStream) returns (TestReply) {}
    // rpc DoubleStream (stream TestRequest) returns (stream TestReply) {}
}

message NumberStream {
    uint64 number = 1;
}
message TestRequest {
    string message = 1;
    int32 counter = 2;
}

message TestReply {
    string message = 1;
    int32 counter = 2;
}
